packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx

```diff
@@ -75,6 +79,46 @@ export const ActionIcon = ({ status }: ActionIconProps): React.JSX.Element => {
 };
 
 const renderRowItem: RenderRowItem<CancelableTask> = (row, index) => {
+  const renderTableData = (
+    columnKey: keyof CancelableTask,
+    row: CancelableTask
+  ) => {
+    switch (columnKey) {
+      case 'key':
+        return (
+          <TableDataText>
+            <ActionIcon status={row.status} />
+            {row.key}
+          </TableDataText>
+        );
+      case 'type': {
+        const indexOfDot = row.key.lastIndexOf('.');
+        return indexOfDot > -1 ? (
+          <TableDataText>
+            {row.key.slice(indexOfDot + 1).toUpperCase()}
+          </TableDataText>
+        ) : (
+          ''
+        );
```

Comment:
Same feedback regarding usage of parsing based on `.` position. Just a general callout that if we have data that is only used for display that we should handle any processing/parsing previous to passing data to consumers to prevent additional work in the react render cycle